### PR TITLE
add KB entry for Array-based batch queries

### PR DESCRIPTION
**Pull request by ybadaoui-ostorlab**
This PR is for adding a KB entry for Array-based batch queries

### CODE REVIEW

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    "title": "Array-Based Batch Queries",
+    "short_description": "Exploiting a GraphQL server by sending array-based batch queries, potentially leading to Denial of Service attacks.",
+    "references": {
+      "Inigo": "https://inigo.io/blog/defeating_controls_with_array-based_query_batching"
+    },
+    "privacy_issue": false,
+    "security_issue": true,
+    "categories": {
+        "CWE_TOP_25": [
+        "CWE_400"
```
Comment (ohachimOs) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json:12]: CWE-20: improper input validation

Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json]: fixed!

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    "title": "Array-Based Batch Queries",
+    "short_description": "Exploiting a GraphQL server by sending array-based batch queries, potentially leading to Denial of Service attacks.",
+    "references": {
+      "Inigo": "https://inigo.io/blog/defeating_controls_with_array-based_query_batching"
+    },
+    "privacy_issue": false,
+    "security_issue": true,
+    "categories": {
+        "CWE_TOP_25": [
+        "CWE_400","CWE_20"
```
Comment (nmasdoufi-ol) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json:12]: I don't think `CWE-400` is part of `CWE-TOP-25`

Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json:12]: accurate! i will remove it, thanks.

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    "privacy_issue": false,
+    "security_issue": true,
+    "categories": {
+        "CWE_TOP_25": [
+        "CWE_20"
+        ],
+        "PCI_STANDARDS": [
+        "REQ_6_2",
+        "REQ_6_4",
+        "REQ_11_3"
```
Comment (deadly-panda) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json:17]: Fix the formatting, please.

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+To mitigate the risk of array-based batch queries attacks in GraphQL, where multiple queries are bundled into a single request, you can apply the following strategies:
+
+Implement Query Complexity Limits: Set rules that account for the number of queries being processed in a single request. This helps limit the load on the server and prevents attackers from sending excessively large batches of queries.
+Limit the Number of Queries Per Request: Enforce server-side restrictions on how many queries can be sent in a single batch. Tools like GraphQL Armor can help impose limits and prevent resource overloading through array-based batch queries.
```
Comment (deadly-panda) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md:4]:
```suggestion
* Implement Query Complexity Limits: Set rules that account for the number of queries being processed in a single request. This helps limit the load on the server and prevents attackers from sending excessively large batches of queries.
* Limit the Number of Queries Per Request: Enforce server-side restrictions on how many queries can be sent in a single batch. Tools like GraphQL Armor can help impose limits and prevent resource overloading through array-based batch queries.
```

Comment (deadly-panda) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md:4]: What's the difference between these two?

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json (hunk 3 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json:17]: fixed!

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md (hunk 4 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md:4]: added, thank you!

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        "V4_1_5",
+        "V5_3_3",
+        "V5_3_4",
+        "V9_1_1",
+        "V9_1_2",
+        "V12_1_1",
+        "V12_1_2"
+        ],
+        "OWASP_ASVS_L3": [
+        "V4_1_3",
```
Comment (oussamaessaji) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json:33]: This one also exists in `OWASP_ASVS_L1` why u didn't include it

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md (hunk 4 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md:4]: query complexity limits restrict the complexity of individual queries, while Limit the Number of Queries Per Request restricts the total number of queries in a single request.

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json (hunk 5 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json:33]: which one!

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        "OWASP_MASVS_L2": [
+        "MSTG_PLATFORM_2"
+        ],
+        "OWASP_ASVS_L2": [
+        "V4_1_3",
+        "V4_1_5",
+        "V5_3_3",
+        "V5_3_4",
+        "V9_1_1",
+        "V9_1_2",
```
Comment (3asm) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json:28]: I think there are too much.

**Review by 3asm — APPROVED** [review]

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json (hunk 6 above)
Comment (ybadaoui-ostorlab) [reply WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/meta.json]: done!

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+To mitigate the risk of array-based batch queries attacks in GraphQL, where multiple queries are bundled into a single request, you can apply the following strategies:
+
+* Implement Query Complexity Limits: Set rules that account for the number of queries being processed in a single request. This helps limit the load on the server and prevents attackers from sending excessively large batches of queries.
```
Comment (elyousfi5) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md:3]: keep it consistent with other KBs

```suggestion
- **Implement Query Complexity Limits**: Set rules that account for the number of queries being processed in a single request. This helps limit the load on the server and prevents attackers from sending excessively large batches of queries.
```

File: WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+* Limit the Number of Queries Per Request: Enforce server-side restrictions on how many queries can be sent in a single batch. Tools like GraphQL Armor can help impose limits and prevent resource overloading through array-based batch queries.
```
Comment (elyousfi5) [inline WEB_SERVICE/WEB/_MEDIUM/ARRAY_BASED_BATCH_QUERIES/recommendation.md:4]:
```suggestion
- **Limit the Number of Queries Per Request**: Enforce server-side restrictions on how many queries can be sent in a single batch. Tools like GraphQL Armor can help impose limits and prevent resource overloading through array-based batch queries.
```

**Review by 3asm — APPROVED** [review]

**Pull request merged**